x-pack/plugins/uptime/server/lib/saved_objects.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  DynamicSettings,
+  defaultDynamicSettings,
+} from '../../../../legacy/plugins/uptime/common/runtime_types/dynamic_settings';
+import { SavedObjectsType } from '../../../../../src/core/server';
+import { UMSavedObjectsQueryFn } from './adapters';
+
+export interface UMDynamicSettingsType {
+  heartbeatIndices: string;
+}
+
+export interface UMSavedObjectsAdapter {
+  getUptimeDynamicSettings: UMSavedObjectsQueryFn<DynamicSettings>;
+  setUptimeDynamicSettings: UMSavedObjectsQueryFn<void, DynamicSettings>;
+}
+
+export const settingsObjectType = 'uptime-dynamic-settings';
+export const settingsObjectId = 'uptime-dynamic-settings-singleton';
+
+export const umDynamicSettings: SavedObjectsType = {
+  name: settingsObjectType,
+  hidden: false,
+  namespaceAgnostic: false,
+  mappings: {
+    properties: {
+      heartbeatIndices: {
+        type: 'keyword',
+      },
+    },
+  },
+};
+
+export const savedObjectsAdapter: UMSavedObjectsAdapter = {
+  getUptimeDynamicSettings: async (client): Promise<DynamicSettings> => {
+    try {
+      const obj = await client.get<DynamicSettings>(umDynamicSettings.name, settingsObjectId);
+      return obj.attributes;
+    } catch (e) {
+      return (
+        await client.create(umDynamicSettings.name, defaultDynamicSettings, {
+          id: settingsObjectId,
+          overwrite: false,
+        })
+      ).attributes;
+    }
```

Comment:
There are many reasons why this could fail. I suspect you only want to attempt recreation if the settings object couldn't be found? Otherwise you risk overwriting the user's configured settings, and since we're not rethrowing from the catch block, the real cause of the failure would likely never be recorded.

Something like this should to the trick. You can import `SavedObjectsErrorHelpers` from `src/core/server`. I'm pretending you have a logger in scope, but you can pull that in from your plugin if you need:

```suggestion
      if (SavedObjectsErrorHelpers.isNotFoundError(e)) {
        log.info('Uptime settings not found. Creating new settings');
        return (
          await client.create(umDynamicSettings.name, defaultDynamicSettings, {
            id: settingsObjectId,
            overwrite: false,
          })
        ).attributes;
      }
      log.error(`Error retrieving uptime settings: ${e}`);
      throw e;
    }
    }
```